mirrord/cli/src/execution.rs

```diff
@@ -362,4 +364,16 @@ impl MirrordExecution {
             .map_err(CliError::InternalProxyWaitError)?;
         Ok(())
     }
+
+    /// Kills the child process, stopping the internal proxy, and completing the agent.
+    ///
+    /// Used when mirrord execution fails inside `execvp`.
+    pub async fn stop(self) {
+        let Self { mut child, .. } = self;
+
+        child
+            .kill()
+            .await
+            .expect("Could not kill intproxy child process!");
+    }
```

Comment:
```suggestion
    pub fn stop(self) {
        let Self { mut child, .. } = self;

        let _ = child.start_kill();
    }
```
We don't really need to wait for process to die and we can't do much in case of failure (e.g. intproxy has already died on its own for some reason)